File: api-service/handlers/health.go
package handlers

import (
	"net/http"

	"api-service/cache"
	"api-service/database"

	"github.com/gin-gonic/gin"
)

func HealthCheck(c *gin.Context) {
	dbHealthy := database.CheckHealth()
	redisHealthy := cache.CheckHealth()

	status := "healthy"
	statusCode := http.StatusOK

	if !dbHealthy || !redisHealthy {
		status = "unhealthy"
		statusCode = http.StatusServiceUnavailable
	}

	c.JSON(statusCode, gin.H{
		"status":   status,
		"database": dbHealthy,
		"redis":    redisHealthy,
	})
}

File: analytics-service/routes/health.py
from fastapi import APIRouter
from services import check_database_health

router = APIRouter()


@router.get("/health")
def health_check():
    """Health check endpoint"""
    db_healthy = check_database_health()
    status = "healthy" if db_healthy else "unhealthy"

    return {
        "status": status,
        "database": db_healthy
    }
